component/azstorage/block_blob.go

```diff
@@ -572,24 +572,32 @@ func (bb *BlockBlob) List(prefix string, marker *string, count int32) ([]*intern
 
 	// For some directories 0 byte meta file may not exists so just create a map to figure out such directories
 	var dirList = make(map[string]bool)
-
 	for _, blobInfo := range listBlob.Segment.BlobItems {
-		attr := &internal.ObjAttr{
-			Path:   split(bb.Config.prefixPath, *blobInfo.Name),
-			Name:   filepath.Base(*blobInfo.Name),
-			Size:   *blobInfo.Properties.ContentLength,
-			Mode:   0,
-			Mtime:  *blobInfo.Properties.LastModified,
-			Atime:  dereferenceTime(blobInfo.Properties.LastAccessedOn, *blobInfo.Properties.LastModified),
-			Ctime:  *blobInfo.Properties.LastModified,
-			Crtime: dereferenceTime(blobInfo.Properties.CreationTime, *blobInfo.Properties.LastModified),
-			Flags:  internal.NewFileBitMap(),
-			MD5:    blobInfo.Properties.ContentMD5,
+		attr := &internal.ObjAttr{}
+		if blobInfo.Properties.CustomerProvidedKeySHA256 != nil && *blobInfo.Properties.CustomerProvidedKeySHA256 != "" {
```

Comment:
Can you please add changelog and tests for this change.

Reply:
For the tests i think the ideal way to go about this would be running all the tests in block_blob_test.go with CPK enabled in config, i think the change might go in pipeline for this. 
Me adding test for only get attribute could lead to cases being missed just like this bug was missed. 
Let me know what you think ?

Reply:
Each test in block_blob_test.go runs with a separate config. So, it might not be straight forward to enable CPK and run those tests. Can you instead add the tests for the use case which was failing.

Reply:
updated.